### PR TITLE
Add tests for error in InfoS

### LINKED ISSUE #158: InfoS uses `%#s` for errors instead of `%q`
/kind bug
When working on https://github.com/kubernetes/kubernetes/pull/91591 we noticed that 
```
klog.V(2).InfoS("NODE DENY", "err", err)
```
was serialized to 
```
I0603 09:57:29.759232   53238 node_authorizer.go:209] "NODE DENY" err=&errors.errorString{s:"node \"node2\" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"}
```
instead of 
```
I0603 09:57:29.759232   53238 node_authorizer.go:209] "NODE DENY" err="node \"node2\" cannot get secret ns1/node1-only, no relationship to this object was found in the node authorizer graph"
```

Possibly it was fixed by https://github.com/kubernetes/klog/issues/156 
If so we should  add tests for Error objects.

### PR DESCRIPTION
**Pull request by tahsinrahman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #158 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: klog_test.go [hunk 1]
```diff
@@ -792,6 +793,16 @@ func TestInfoS(t *testing.T) {
 			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" replicaNum=20\n",
 			keysValues: []interface{}{"replicaNum", 20},
 		},
+		{
+			msg:        "test",
+			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" err=test error\n",
```
Comment (serathius) [inline klog_test.go:798]:
```suggestion
			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" err="test error"\n",
```

File: klog_test.go [hunk 2]
```diff
@@ -792,6 +793,16 @@ func TestInfoS(t *testing.T) {
...
+			keysValues: []interface{}{"err", errors.New("test error")},
+		},
+		{
+			msg:        "test",
+			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" err=test error\n",
```
Comment (serathius) [inline klog_test.go:803]:
```suggestion
			format:     "I0102 15:04:05.067890    1234 klog_test.go:%d] \"test\" err="test error"\n",
```

**Comment by serathius** [comment]
errors should be formatted using `%q` not `%s`. Difference is additional quotes and escaping.

File: klog_test.go (hunk 1 above)
Comment (tahsinrahman) [reply klog_test.go]: got it, updated!

File: klog.go [hunk 3]
```diff
@@ -804,6 +804,8 @@ func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
 		switch v.(type) {
 		case string:
 			b.WriteString(fmt.Sprintf("%s=%q", k, v))
+		case error:
```
Comment (yuzhiquan) [inline klog.go:807]: How about
```
case string, error:
	   b.WriteString(fmt.Sprintf("%s=%q", k, v)
```

**Pull request merged**